Classes/URKArchive.mm

```diff
@@ -331,7 +403,32 @@ - (nullable NSURL *)firstVolumeURL {
 
 - (nullable NSArray<NSString*> *)listVolumePaths:(NSError **)error
 {
-    return @[];
+    __block NSMutableArray<NSString*> *volumePaths = [NSMutableArray new];
+    
+    URKArchive * dummy = self;
+    NSURL * firstVolumeURL = [self firstVolumeURL];
+    if (firstVolumeURL != self.fileURL)
+    {
+        NSError *firstVolumeURLError = nil;
+        
+        dummy = [self initWithURL:firstVolumeURL error:&firstVolumeURLError];
```

Comment:
We need to find a different solution here. You should never call [self init...] from a Cocoa class. What error are you checking for? This only returns an error when it can’t create a bookmark. What do you need to create the bookmark over again for?